### DISCUSSION TITLE
How to make slash commands work without `nvim-cmp`?

### DISCUSSION THREAD

**Discussion by ecosse3** [olimorris/codecompanion.nvim#501] [Q&A] [answered]
Hi,

I have and issue that any slash commands doesn't work for me so I basically missing the best features of that plugin since I cannot refer buffers/files and any other context. That's because as my completion provider I use great [blink.cmp](https://github.com/Saghen/blink.cmp) plugin which unfortunately isn't supported by Code Companion. I don't want to install nvim-cmp.

I've changed my provider for slash commands to `telescope` but it doesn't trigger anyway. That's the video below.

https://github.com/user-attachments/assets/41bcfb5f-30f4-47cc-b0f8-9d5f2a71dc17

Should I open some issue or is they a way to trigger slash commands with native nvim completion or blink.cmp since it uses compatibility mode for cmp plugins via [blink.compat](https://github.com/Saghen/blink.compat) plugin?

**Comment by olimorris** [comment]
Hey @ecosse3. I've documented this in the [Quickstart](https://github.com/olimorris/codecompanion.nvim?tab=readme-ov-file#rocket-quickstart) section of the README, albeit, it's not too clear:

> These have been designed to work with native Neovim completions and also with [nvim-cmp](https://github.com/hrsh7th/nvim-cmp). To open the completion menu use <C-_> in insert mode when in the chat buffer.

Can you let me know if that works?

**Reply by ecosse3** [reply]
Yes, i've read it but unfortunately I don't use native autocompletion or it doesn't work for me. When I try `<C-x><C-o` in Insert mode to enable autocompletion after I write `/buffer` it shows me `E764: Option 'omnifunc' is not set` even if I set it via `vim.opt.omnifunc = ""` (defaults to empty string). I don't really need autocompletion as I configured slash command provider to `telescope` so I except to open telescope after I type slash command like `/buffer`, but it doesn't work.

I also set `omnifunc = v:lua.vim.lsp.omnifunc` which doesn't give me that error anymore but it also doesn't autocomplete.

**Reply by olimorris** [reply]
What about if you use the keymap that CodeCompanion defines for completion in the chat buffer?

I don't have any omnifunc settings in my dotfiles and it works fine for me although I remapped the keymap to `<C-x>`

**Reply by olimorris** [reply]
@ecosse3 - Sharing this thread for reference: https://github.com/olimorris/codecompanion.nvim/discussions/381

**Reply by ecosse3** [reply]
I think the issue is that the `lua local has_cmp, _ = pcall(require, "nvim-cmp"); print(has_cmp);` is returning `true` in my case, even if I don't have any `nvim-cmp` dependency in my whole nvim config (checked with ripgrep) as well as I don't have `nvim-cmp` directory in my local lazy.nvim folder with plugins. I'll try to investigate why that happens.

**Reply by olimorris** [reply]
> I think the issue is that the `lua local has_cmp, _ = pcall(require, "nvim-cmp"); print(has_cmp);` is returning `true` in my case, even if I don't have any `nvim-cmp` dependency in my whole nvim config (checked with ripgrep) as well as I don't have `nvim-cmp` directory in my local lazy.nvim folder with plugins. I'll try to investigate why that happens.

I'd recommend trying out the minimal.lua config in the repo, disabling cmp, and see if you can work back from that.

**Reply by ecosse3** [reply]
@olimorris I was able to make it work with minimal config. The issue in my config was that I had `saghen/blink.compat` as dependencies of `saghen/blink.cmp` which probably uses nvim-cmp under the hood. Removing that fixes the issue that I have no more nvim-cmp imported in my config. Default mapping `<C-_>` doesn't work but after I changed it to `<C-x>` it works.

I still have issue that I cannot trigger completion to "pick" that slash command but it's probably related to my nvim config. I'll try to find what's wrong. Clicking `<C-x>` again doesn't pick it. I have `Tab` mapped for copilot completion since it doesn't work with blink.cmp yet.

https://github.com/user-attachments/assets/4cadb5e4-713c-4512-b5c0-e14b2bf4bb9c

**Reply by olimorris** [reply]
That's a great spot. Right I'll look to fix this.

**Reply by olimorris** [reply, marked as answer]
@ecosse3 this should be fixed in the latest commit

**Comment by olimorris** [comment]
Btw, blink support is on the way but there's an outstanding issue that I raised with the maintainer last night so the blocker is on their side 😉.
